presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -436,21 +443,24 @@ std::unique_ptr<TaskInfo> TaskManager::createOrUpdateTask(
         return std::make_unique<TaskInfo>(prestoTask->updateInfoLocked());
       }
 
-      execTask = exec::Task::create(
-          taskId, planFragment, prestoTask->id.id(), std::move(queryCtx));
-      auto baseSpillDir = *(baseSpillDir_.rlock());
-      maybeSetupTaskSpillDirectory(planFragment, *execTask, baseSpillDir);
+      {
+        auto newVeloxTask = exec::Task::create(
```

Comment:
Why do we need this variable?

Reply:
This is to destroy the velox task under presto task lock so as to ensure the second presto task update call won't see the leftover state from the previous failed velox task such as the task memory pool under query root memory pool. Here is the event sequence:
T1: task manager receive request from coordinator to create velox task;
T2: task manager holds the presto task lock to create a new velox task;
T3: the spilling directory setup fails and throw, but the new velox task hans't destroyed yet;
T4: task manager releases the presto task lock and second create velox task request comes;
T5: task manager creates a new velox task for the same presto task but the memory pool of the previously created velox task still stays in the query root memory pool;
T6: the second velox task creation fails because of the duplicate task memory pool name.

Reply:
I removed it as maybeSetupDirectory now tolerates the failure.